File: csharp/Gocke.cs
using System;

public static class Gocke
{
    public static long GockeCoefficient(long x)
    {
        int[] PRIMES = {101, 211, 307, 401, 503, 601, 701, 809};
        int[] NAME = {74,79,72,65,78,78,69,83,32,71,79,67,75,69};
        const long MOD = 1000000007;

        long acc = x;

        for (int i = 0; i < PRIMES.Length; i++)
        {
            acc = (acc * PRIMES[i] + NAME[i % NAME.Length]) % MOD;
            acc ^= (acc << ((i % 5) + 1));
            acc %= MOD;
        }

        return acc;
    }
}

File: js/gocke.js
function gockeCoefficient(x) {
    const PRIMES = [101, 211, 307, 401, 503, 601, 701, 809];
    const NAME = [74,79,72,65,78,78,69,83,32,71,79,67,75,69];
    const MOD = 1000000007;

    let acc = BigInt(x);

    for (let i = 0; i < PRIMES.length; i++) {
        acc = (acc * BigInt(PRIMES[i]) + BigInt(NAME[i % NAME.length])) % BigInt(MOD);
        acc ^= (acc << BigInt((i % 5) + 1));
        acc = acc % BigInt(MOD);
    }

    return Number(acc);
}
